### DISCUSSION TITLE
can I build this on Steam OS (arch linux) somehow ?

### DISCUSSION THREAD

**Discussion by iliasch-dev** [input-leap/input-leap#1814] [Q&A]
i've seen that there is a barrier version for this distro but when i tried to install the client and connect it to input-leap server I'm getting an error on SSL handshake which makes me believe barrier is build with a different SSL lib. Can I build input-leap from this repository on arch linux? Anyone done it ?

**Comment by shymega** [comment]
I don't own a Steam Deck, so I can't help you. Try following the wiki guide for Linux, and adjust to take into account SteamOS's immutable filesystem.

Also, Barrier and Input Leap aren't going to be interoperable forever. I don't recommend relying on Barrier mixed with Input Leap.

**Comment by tianon** [comment]
Honestly a flatpak (and thus https://github.com/input-leap/input-leap/discussions/1652) is probably a more sustainable avenue for getting this to work 👀

**Comment by iliasch-dev** [comment]
i checked the wiki and it's very confusing. I have run 
`flatpak-builder build-dir org.inputleap.InputLeap.json --force-clean`
to build the flatpak but i get the error 

 `flatpak-builder-ERROR **: 10:23:29.169: Failed to load included manifest (/home/ilias/git/input-leap/shared-modules/intltool/intltool-0.51.json): Failed to open file ?/home/ilias/git/input-leap/shared-modules/intltool/intltool-0.51.json?: No such file or directory`

there's no such file. However the org.inputleap.InputLeap.json the wiki mentions for download, references that file.  

https://github.com/flathub/com.github.debauchee.barrier/blob/master/com.github.debauchee.barrier.json

**Reply by shymega** [reply]
The wiki is outdated. Also, that file is for Barrier - it will not work with Input Leap. Flatpak is on our list of items to do, but it's not a priority at this time.

Currently, your best bet is to use the AUR package for Input Leap along with the `libportal` package with the `InputCapture` PR applied, as the Steam Deck runs Arch. That's all I can advise for now.

**Reply by shymega** [reply]
I've updated the wiki to strikethrough those old instructions. There was a note to tell people *not* to use those, but clearly, it wasn't clear enough.

**Comment by karypid** [comment]
Have you managed to build the flatpak? I am on arch-based EndeavourOS and would like to try this...

Which wiki did you use to build it?

**Comment by joaociocca** [comment]
~hi! considering that flatpak build has been released, how can I run the client from it in CLI/systemd service~

~I tried stuff like `flatpak run io.github.input_leap.InputLeap -f --no-tray --debug INFO --name steamdeck --disable-crypto <server_ip>`, but it just pops the client GUI, while the terminal shows an error: `Qt: Session management error: Could not open network socket`.~

~After it, if I insert the server IP and click "Start", the terminal shows `QList("-f", "--no-tray", "--debug", "INFO", "--name", "steamdeck", "--disable-crypto", "[<server_ip>]:24800")` and connects to the server.~

~But I can't seem to use the flatpak to don't show the GUI and just connect to the server.~

found out how to do it: `/usr/bin/flatpak run --command=/app/bin/input-leapc io.github.input_leap.InputLeap --name steamdeck --disable-crypto -f --debug INFO <server_ip>`
